c2/c2-client-bundle/c2-client-http/src/main/java/org/apache/nifi/c2/client/http/url/C2UrlProviderFactory.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.c2.client.http.url;
+
+import static org.apache.commons.lang3.StringUtils.isNoneBlank;
+
+import org.apache.nifi.c2.client.C2ClientConfig;
+
+public class C2UrlProviderFactory {
+
+    private final C2ClientConfig clientConfig;
+
+    public C2UrlProviderFactory(C2ClientConfig clientConfig) {
+        this.clientConfig = clientConfig;
+    }
+
+    public C2UrlProvider create() {
+        if (isNoneBlank(clientConfig.getC2RestApi(), clientConfig.getC2RestPathHeartbeat(), clientConfig.getC2RestPathAcknowledge())) {
+            return new ProxyAwareC2UrlProvider(clientConfig.getC2RestApi(), clientConfig.getC2RestPathHeartbeat(), clientConfig.getC2RestPathAcknowledge());
+        } else if (isNoneBlank(clientConfig.getC2Url(), clientConfig.getC2AckUrl())) {
+            return new LegacyC2UrlProvider(clientConfig.getC2Url(), clientConfig.getC2AckUrl());
+        } else {
+            throw new IllegalArgumentException("Incorrect configuration. Please configure legacy or proxy aware C2 URL properties");
```

Comment:
I think it would make sense to provide a little more information here I don't think all users would understand what legacy / proxy aware C2 URL properties means. Listing explicitly all the properties for legacy and for proxy aware would be a big help I suppose. What do you think?

Reply:
Added a more detailed error message